analytics/analytics/tests/test_upload_error_taxonomy.py

```diff
@@ -0,0 +1,339 @@
+                    "name": "Spack Bot",
+                    "username": "spackbot",
+                    "avatar_url": "https://gitlab.spack.io/uploads/-/system/user/avatar/3/57643519.png",
+                    "email": "[REDACTED]",
+                },
+                "commit": {
+                    "id": 547935,
+                    "name": None,
+                    "sha": "a0d0305f2ff9a9f375010aaa79336d8241b9ab0f",
+                    "message": "Merge 78c2eb5afb669bd28c52ae68d13ef49f95432227 into 4027a2139b053251dafc2de38d24eac4d69d42a0\n",
+                    "author_name": "spackbot",
+                    "author_email": "[REDACTED]",
+                    "author_url": "mailto:noreply@spack.io",
+                    "status": "running",
+                    "duration": None,
+                    "started_at": "2023-11-10 14:44:09 UTC",
+                    "finished_at": None,
+                },
+                "repository": {
+                    "name": "spack",
+                    "url": "git@ssh.gitlab.spack.io:spack/spack.git",
+                    "description": "",
+                    "homepage": "https://gitlab.spack.io/spack/spack",
+                    "git_http_url": "https://gitlab.spack.io/spack/spack.git",
+                    "git_ssh_url": "git@ssh.gitlab.spack.io:spack/spack.git",
+                    "visibility_level": 20,
+                },
+                "project": {
+                    "id": 2,
+                    "name": "spack",
+                    "description": "",
+                    "web_url": "https://gitlab.spack.io/spack/spack",
+                    "avatar_url": None,
+                    "git_ssh_url": "git@ssh.gitlab.spack.io:spack/spack.git",
+                    "git_http_url": "https://gitlab.spack.io/spack/spack.git",
+                    "namespace": "spack",
+                    "visibility_level": 20,
+                    "path_with_namespace": "spack/spack",
+                    "default_branch": "develop",
+                    "ci_config_path": "share/spack/gitlab/cloud_pipelines/.gitlab-ci.yml",
+                },
+                "environment": None,
+            },
+            "concretization_error",
+        ),
+        # TODO: this test case fails.
+        # (
+        #     {
+        #         "before_sha": "a6466b9dddf59cc185800ac428bd4ba535b96c2e",
+        #         "build_id": 8826259,
+        #         "build_name": "py-scipy@1.5.4 /mscpuhz %gcc@11.3.0 arch=linux-ubuntu22.04-x86_64_v3 Machine Learning",
+        #         "build_stage": "stage-13",
+        #         "build_status": "failed",
+        #         "object_kind": "build",
+        #         "ref": "develop",
+        #         "retries_count": 1,
+        #         "sha": "a452e8379e12bd46925df30b99cf4b30edf80457",
+        #         "tag": False,
+        #     },
+        #     "other",
+        # ),
+    ],
+)
+@pytest.mark.django_db()
+def test_upload_gitlab_failure_logs(
```

Comment:
Should we setup a github actions workflow to run these tests? It would require pretty minimal config afaik.

Reply:
https://github.com/spack/spack-infrastructure/pull/698/commits/156f91ef79840050b2bcbde51805359ddf54a2a7